cpp/src/io/csv/reader_impl.cu

```diff
@@ -859,7 +867,7 @@ table_with_metadata read_csv(cudf::io::datasource* source,
         const std::string dblquotechar(2, parse_opts.quotechar);
         std::unique_ptr<column> col = cudf::make_strings_column(*out_buffers[i]._strings, stream);
         out_columns.emplace_back(
-          cudf::strings::replace(col->view(), dblquotechar, quotechar, -1, mr));
+          cudf::strings::detail::replace(col->view(), dblquotechar, quotechar, -1, stream, mr));
```

Comment:
Unrelated change, just noticed that the version that the detail API needs to be used here to ensure that the right stream is used.